Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSessionCocoa.mm

```diff
@@ -263,17 +277,21 @@ static void didReceiveServerTrustChallenge(Ref<NetworkConnectionToWebProcess>&&
 #endif // HAVE(WEB_TRANSPORT)
 }
 
-void NetworkTransportSession::sendDatagram(std::span<const uint8_t> data, CompletionHandler<void()>&& completionHandler)
+void NetworkTransportSession::sendDatagram(std::span<const uint8_t> data, CompletionHandler<void(std::optional<WebCore::Exception>&&)>&& completionHandler)
 {
 #if HAVE(WEB_TRANSPORT)
     ASSERT(m_datagramConnection);
     nw_connection_send(m_datagramConnection.get(), makeDispatchData(Vector(data)).get(), NW_CONNECTION_DEFAULT_MESSAGE_CONTEXT, true, makeBlockPtr([completionHandler = WTFMove(completionHandler)] (nw_error_t error) mutable {
-        ASSERT(!error);
-        // FIXME: Pass any error through to JS.
-        completionHandler();
+        if (error) {
+            if (!(nw_error_get_error_domain(error) == nw_error_domain_posix && nw_error_get_error_code(error) == ECANCELED))
+                completionHandler(WebCore::Exception(WebCore::ExceptionCode::NetworkError));
+            // If we were cancelled, exit gracefully
+            return;
```

Comment:
If it's not a posix cancelled error, we won't ever call the completionHandler, which will assert.  It looks like we're missing a completionHandler(std::nullopt)